cfme/fixtures/cli.py

```diff
@@ -52,18 +53,28 @@ def fqdn_appliance(appliance):
     else:
         logger.error("Couldn't provision an appliance at all")
         raise SproutException('No provision available')
-    yield apps[0]
+    return apps[0]
```

Comment:
**REQUIRED:** The code beneath this is unreachable, once you return it will never be run

Reply:
i propose using `contextlib.contextmanager` for the helper function and then using with statements in the fixtures in order to manage the state,

that way the yield can be used and the rest of the code will still do cleanup